### DISCUSSION TITLE
AKHQ Filter Topics By Connection

### DISCUSSION THREAD

**Discussion by josue-sanchez-wabtec** [tchiotludo/akhq#1536] [Q&A]
Is there any possibility to filter topics by connection and using the same service account at sasl.jaas.config ?

Lets say I have a hundred connections configured, each one with a different schema registry section like: 

`
Feature-1:
      schema-registry:
        url: https://feature-1.schema.registry:443/schema-registry
        type: confluent
        basic-auth-username: <some_user>
        basic-auth-password: <some_password>
      ui-options:
        topic-data:
          sort: NEWEST
      properties:
        bootstrap.servers: server-url.confluent.cloud:9092
        security.protocol: SASL_SSL
        sasl.mechanism: PLAIN
        sasl.jaas.config: org.apache.kafka.common.security.plain.PlainLoginModule required username='some_SA_user' password='some_SA_password';

Feature-2:
      schema-registry:
        url: https://feature-2.schema.registry:443/schema-registry
        type: confluent
        basic-auth-username: <some_user>
        basic-auth-password: <some_password>
      ui-options:
        topic-data:
          sort: NEWEST
      properties:
        bootstrap.servers: server-url.confluent.cloud:9092
        security.protocol: SASL_SSL
        sasl.mechanism: PLAIN
        sasl.jaas.config: org.apache.kafka.common.security.plain.PlainLoginModule required username='some_SA_user' password='some_SA_password';

...

Feature-100:
...
  `

All of these envs handles topics prefixed like:

dev-topic-feature-1-service-1
dev-topic-feature-1-service-2
dev-topic-feature-1-service-3
...
dev-topic-feature-1-service-40
dev-topic-feature-2-service-1
dev-topic-feature-2-service-2
dev-topic-feature-2-service-3
...
dev-topic-feature-2-service-40
.
.
.
dev-topic-feature-100-service-40

  
I have the same sasl.jaas.config configuration for all of the connections which has the Service Account that handles all the topics with a prefix "dev-topic-feature". 

What I see is that all the connections get the same amount of topics following the prefix rule "dev-topic-feature", but I want only the corresponding connection Feature-# topics to be selected on each connection with the context already explain.

![image](https://github.com/tchiotludo/akhq/assets/114017629/d6e02bca-0798-4173-a3d4-a53754679c6a)

I know there is a solution for this changing all Service Accounts from sasl.jaas.config to a specific prefixed one.

Lets say:

For Feature-1 I can create a sasl.jaas.config Service Account with prefix permission for "dev-topic-feature-1"
For Feature-2 I can create a sasl.jaas.config Service Account with prefix permission for "dev-topic-feature-2"

So the filter will happen, but this mean I need to create n amount of Service Accounts.

Is there any akhq configuration option you could recommend me? Another approach perhaps?

Thank you in advance.

**Comment by tchiotludo** [comment]
Try the latest dev version that could allow this kind of filtering as I understand
